libsolidity/ast/AST.h

```diff
@@ -258,10 +258,16 @@ class Declaration: public ASTNode, public Scopable
 	bool isVisibleAsLibraryMember() const { return visibility() >= Visibility::Internal; }
 	virtual bool isVisibleViaContractTypeAccess() const { return false; }
 
-
 	virtual bool isLValue() const { return false; }
 	virtual bool isPartOfExternalInterface() const { return false; }
 
+	/// @returns true if this is a declaration of an enum member.
+	bool isEnumValue() const;
+	/// @returns true if this is a declaration of a struct member.
+	bool isStructMember() const;
```

Comment:
See https://github.com/ethereum/solidity/pull/9408/files#diff-31ae3d0bfd1c2c77c869a76b47c9bdfdR943-R944, but placed in `Declaration`.

Reply:
Why here instead of in `VariableDeclaration` which already contains similar helpers?

Reply:
Yes, I started with extending `VariableDeclaration` :). But consider this [check](https://github.com/ethereum/solidity/pull/9903/files#diff-653a1e23aa75715b13289b03e2f20947R461)
```
bool warnOnShadow = !_declaration.isStructMember() && !_declaration.isEnumValue() && !_declaration.isEventParameter();
```
The parameter `_declaration` has type `Declaration`. It is slightly more convenient to avoid casting to `VariableDeclaration` before calling `isStructMember()` and `isEventParameter()`.

But more important is that `EnumValue` is not `VariableDeclaration`. So we could not use `isEnumValue()`, and would need a yet another cast.